tests/unit/dataactvalidator/test_c9_award_financial.py

```diff
@@ -1,37 +1,151 @@
+    af = AwardFinancialFactory(
+        tas=tas, submisson_id=afa.submission_id, fain='xyz', uri=None)
+
+    errors = number_of_errors(_FILE, database, models=[afa, af])
+    assert errors == 1
+
+
+def test_unequal_uri(database):
+    """Tests File D2 (award financial assistance) uri different than File C (award financial) uri."""
+    tas = _TAS
+    afa = AwardFinancialAssistanceFactory(
+        tas=tas, fain=None, uri='abc', federal_action_obligation='1', original_loan_subsidy_cost='1')
+    af = AwardFinancialFactory(
+        tas=tas, submisson_id=afa.submission_id, fain=None, uri='xyz')
+
+    errors = number_of_errors(_FILE, database, models=[afa, af])
+    assert errors == 1
+
+
+def test_unequal_fain_null(database):
+    """Tests non-NULL File D2 (award financial assistance) fain compared to a NULL
+    fain in File C (award financial)."""
+    tas = _TAS
+    afa = AwardFinancialAssistanceFactory(
+        tas=tas, fain='abc', uri=None, federal_action_obligation='1', original_loan_subsidy_cost='1')
+    af = AwardFinancialFactory(
+        tas=tas, submisson_id=afa.submission_id, fain=None, uri=None)
+
+    errors = number_of_errors(_FILE, database, models=[afa, af])
+    assert errors == 1
+
+
+def test_unequal_uri_null(database):
+    """Tests NULL File D2 (award financial assistance) uri compared to a non-NULL
+    uri in File C (award financial)."""
+    tas = _TAS
+    afa = AwardFinancialAssistanceFactory(
+        tas=tas, fain=None, uri=None, federal_action_obligation='1', original_loan_subsidy_cost='1')
+    af = AwardFinancialFactory(
+        tas=tas, submisson_id=afa.submission_id, fain=None, uri='abc')
+
+    errors = number_of_errors(_FILE, database, models=[afa, af])
+    assert errors == 1
+
+
+def test_zero_federal_action_obligation_and_original_loan_subsidy_cost(database):
+    """Tests that a single warning is thrown for both a federal action obligation of 0
+     and an original loan subsidy cost of 0"""
+    tas = _TAS
+    afa = AwardFinancialAssistanceFactory(
+        tas=tas, fain='abc', uri=None, federal_action_obligation='0', original_loan_subsidy_cost='0')
+    af = AwardFinancialFactory(
+        tas=tas, submisson_id=afa.submission_id, fain=None, uri=None)
+
+    errors = number_of_errors(_FILE, database, models=[afa, af])
+    assert errors == 0
+
+
+def test_zero_federal_action_obligation_and_positive_original_loan_subsidy_cost(database):
+    """Tests that a single warning is thrown for both a federal action obligation of 0
+     and an original loan subsidy cost of 0"""
+    tas = _TAS
+    afa = AwardFinancialAssistanceFactory(
+        tas=tas, fain='abc', uri=None, federal_action_obligation='0', original_loan_subsidy_cost='1')
+    af = AwardFinancialFactory(
+        tas=tas, submisson_id=afa.submission_id, fain=None, uri=None)
+
+    errors = number_of_errors(_FILE, database, models=[afa, af])
+    assert errors == 0
```

Comment:
I'm surprised that we don't expect this one to error, that fain is not present in C and the fields are not both 0, what am I missing?